packages/surveysparrow_flutter_sdk/lib/surveysparrow.dart

```diff
@@ -0,0 +1,749 @@
+      String phoneValue = ""}) {
+    createAnswerPayload(
+      _collectedAnswers,
+      key,
+      value,
+      _surveyToMap,
+      otherInput,
+      otherInputText,
+      otherInputId,
+      isPhoneInput,
+      phoneValue,
+      (_stopwatch.elapsedMilliseconds / 1000).round(),
+    );
+    var _cloneWorkBench = {..._workBench};
+
+    var newWorkBench = getWorkBenchData(
+      _cloneWorkBench,
+      key,
+      value,
+      otherInputText,
+      _answeredCount,
+      otherInput,
+      isPhoneInput,
+      phoneValue,
+    );
+
+    setState(() {
+      _workBench = {...newWorkBench};
+    });
+
+    widget.currentlyCollectedAnswers!(_collectedAnswers);
+
+    Future.delayed(const Duration(milliseconds: 500), () {
+      _handleNextQuestion();
+    });
+
+    if (_lastQuestion['id'] == key) {
+      widget.allCollectedAnswers!(_collectedAnswers);
+    }
+  }
+
+  submitData() async {
+    if (hasThankYouPage) {
+      setState(() {
+        _pageType = 'thankYou';
+      });
+    } else {
+      this.widget.onSubmitCloseModalFunction!();
+    }
+
+    var data = await submitAnswer(
+        _collectedAnswers,
+        (_stopwatch.elapsedMilliseconds / 1000).round(),
+        customParams,
+        this.token,
+        this.widget.domain);
+  }
+
+  _scrollListener() {
+    if (_scrollControler.offset >= _scrollControler.position.maxScrollExtent &&
+        !_scrollControler.position.outOfRange) {
+      _scrollCountBottom += 1;
+      _scrollCountTop = 0;
+      if (_scrollCountBottom == 2) {
+        _handleNextQuestion();
+        _scrollCountTop = 0;
+        _scrollCountBottom = 0;
+      }
+    }
+
+    if (_scrollControler.offset <= _scrollControler.position.minScrollExtent &&
+        !_scrollControler.position.outOfRange) {
+      _scrollCountTop += 1;
+      _scrollCountBottom = 0;
+      if (_scrollCountTop == 2) {
+        _handlePreviousQuestion();
+        _scrollCountTop = 0;
+        _scrollCountBottom = 0;
+      }
+    }
+
+    if (_scrollControler.offset < _scrollControler.position.maxScrollExtent &&
+        _scrollControler.offset > _scrollControler.position.minScrollExtent &&
+        !_scrollControler.position.outOfRange) {
+      _scrollCountTop = 0;
+      _scrollCountBottom = 0;
+    }
+  }
+
+  late Stopwatch _stopwatch;
+  late ScrollController _scrollControler;
+
+  var hasWelcomePage = false;
+  Map<dynamic, dynamic> welcomePageDetails = {};
+  var welcomeDesc = "";
+  var welcomeButtonDesc = "";
+  Map<dynamic, dynamic> welcomeEntity = {};
+
+  var isVertical = false;
+
+  // thank you json
+  var hasThankYouPage = false;
+  var thankYouPageJson = {};
+
+  incrementCount(token, domain) async {
+    var url =
+        Uri.parse('http://${domain}/api/internal/sdk/increment-count/${token}');
+    final response = await http.get(url);
+  }
+
+  @override
+  initState() {
+    super.initState();
+    incrementCount(token, this.widget.domain);
+
+    if (widget.euiTheme!['bottomSheet'] != null) {
```

Comment:
@sachin-kumar56 , Can we maintain all this keys like `bottomSheet`, `welcome_rtxt`, `welcomeScreenYesButtonText` etc.. in a constant object and then refer them here? This will be useful for us in the long run as we will have control over the places we will be using them.